### PR TITLE
Introduce scheduler CPU/Memory profile-gathering in density test

### PR DESCRIPTION
**Pull request by shyamjvs**
This should help us get more reliable/realistic data for scheduler (from our real-cluster scalability tests).

/cc @wojtek-t 
fyi - @davidopp @bsalamat @misterikkit 

```release-note
NONE
```

### CODE REVIEW

**Comment by shyamjvs** [comment]
@kubernetes/sig-scheduling-misc @kubernetes/sig-scalability-misc

File: test/e2e/framework/profile_gatherer.go [hunk 1]
```diff
@@ -124,67 +153,53 @@ func gatherProfileOfKind(profileBaseName, kind string) error {
...
+// CPU profiles available below.
 
-func GatherApiserverCPUProfile(wg *sync.WaitGroup, profileBaseName string) {
-	GatherApiserverCPUProfileForNSeconds(wg, profileBaseName, DefaultCPUProfileSeconds)
+func GatherCPUProfile(componentName string, profileBaseName string, wg *sync.WaitGroup) {
+	GatherCPUProfileForNSeconds(componentName, profileBaseName, DefaultCPUProfileSeconds, wg)
 }
 
-func GatherApiserverCPUProfileForNSeconds(wg *sync.WaitGroup, profileBaseName string, n int) {
+func GatherCPUProfileForNSeconds(componentName string, profileBaseName string, N int, wg *sync.WaitGroup) {
```
Comment (wojtek-t) [inline test/e2e/framework/profile_gatherer.go]: s/ForNSeconds/ForSeconds/
s/N/seconds/

?

Comment (wojtek-t) [reply test/e2e/framework/profile_gatherer.go]: BTW - this function doesn't seem to be called from anywhere externally - so let's make that private one

File: test/e2e/scalability/density.go [hunk 2]
```diff
@@ -64,6 +64,9 @@ var MaxContainerFailures = 0
 // Maximum no. of missing measurements related to pod-startup that the test tolerates.
 var MaxMissingPodStartupMeasurements = 0
 
+// Number of nodes in the cluster (computed inside BeforeEach).
+var NodeCount = 0
```
Comment (wojtek-t) [inline test/e2e/scalability/density.go]: This shouldn't be public.

s/NodeCount/nodeCount/

Comment (shyamjvs) [reply test/e2e/scalability/density.go]: Done

File: test/e2e/framework/profile_gatherer.go (hunk 1 above)
Comment (shyamjvs) [reply test/e2e/framework/profile_gatherer.go]: > s/ForNSeconds/ForSeconds/ s/N/seconds/

Done

> BTW - this function doesn't seem to be called from anywhere externally - so let's make that private one

I'd prefer keeping it public - to not force users of this utility (which although is just us currently) to only be able to use polling-based gatherer. Wdyt?

**Comment by wojtek-t** [comment]
@bsalamat - e.g. this one:
http://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/14940/artifacts/profiles/
[this one is for 100-node cluster, but it shows the pattern]

You access testgrid, and it's in artifacts/profiles directory for a given test run.

**Pull request merged**